src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTestCS.java

```diff
@@ -0,0 +1,55 @@
+package nl.hsac.fitnesse.fixture.slim;
+
+
+import java.util.Map;
+import org.apache.http.client.CookieStore;
+import org.apache.http.cookie.Cookie;
+import org.apache.http.impl.client.BasicCookieStore;
+import nl.hsac.fitnesse.fixture.util.HttpResponse;
+
+
+
+
+
+/**
+ * Fixture to make HTTP requests (With multiple cookies support) using Slim scripts and/or scenarios.
+ */
+public class HttpTestCS extends HttpTest {
+   
+	
+	/**
+     * @return headers received with response to last request.
+     */
+	
+	
+    public Map<String, String> responseHeaders() {
+    	
+    	HttpResponse response = super.getResponse();
+    	Map<String, String> responseHeaders = super.responseHeaders();
+    	StringBuilder sb = new StringBuilder();
+    	CookieStore store = response.getCookieStore();
+    	
+    	if(store!=null)
+    	{
+	    	for(Cookie cookie : store.getCookies())
+	    	{
+	    		sb.append(cookie.getName() + "=" + cookie.getValue());
+	    		sb.append(";");
+	    	}
+    	}
+    	
+    	responseHeaders.put("Set-Cookie", sb.toString());
```

Comment:
I suggest that you only read the cookie store, and put a header, if the responseHeaders contains Set-Cookie, **and the store is not null**.